code/game/machinery/doors/door.dm

```diff
@@ -157,6 +157,9 @@ var/list/all_doors = list()
 	if (istype(I, /obj/item/device/detective_scanner))
 		return
 
+	if(isobserver(user) && !isAdminGhost(user))
```

Comment:
Any chance we can make a macro of this sometime in the future for isghost?

Lot of machines, such as APCs and the cargo console, that don't allow for admin ghost intervention or observation